### PR TITLE
V7: rename "report" to "event"

### PR DESCRIPTION
**Pull request by bengourley**
The construct represented by the "report" class in this codebase is now called "event" in the spec. This also now matches up with the concept of an "event" in our dashboard and other areas of the product.

There are changes to be made to its interface, but these will be included in subsequent PRs.

No functionality should have been changed here, literally just the naming.

### CODE REVIEW

File: packages/plugin-react-native-global-error-handler/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "author": "Bugsnag",
   "license": "MIT",
   "devDependencies": {
-    "@bugsnag/core": "^7.0.0-pre-alpha-ben.6",
+    "@bugsnag/core": "^6.4.1",
```
Comment (djskinner) [inline packages/plugin-react-native-global-error-handler/package.json]: Should this be part of this PR?

File: packages/plugin-react-native-unhandled-rejection/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
   "author": "Bugsnag",
   "license": "MIT",
   "devDependencies": {
-    "@bugsnag/core": "^7.0.0-pre-alpha-ben.6",
+    "@bugsnag/core": "^6.4.1",
```
Comment (djskinner) [inline packages/plugin-react-native-unhandled-rejection/package.json]: Should this be part of this PR?

File: packages/plugin-react-native-unhandled-rejection/test/rejection-handler.test.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ describe('plugin: react native rejection handler', () => {
     }
     stop()
 
-    // the rejection tracker waits 100ms before reporting TypeError as unhandled
+    // the rejection tracker waits 100ms before eventing TypeError as unhandled
```
Comment (djskinner) [inline packages/plugin-react-native-unhandled-rejection/test/rejection-handler.test.js]: Check wording

File: packages/plugin-server-session/session.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ module.exports = {
 const sendSessionSummary = client => sessionCounts => {
   const releaseStage = inferReleaseStage(client)
 
-  // exit early if the reports should not be sent on the current releaseStage
+  // exit early if the events should not be sent on the current releaseStage
```
Comment (djskinner) [inline packages/plugin-server-session/session.js]: is this session not sent or event(s) not sent?

**Comment by bengourley** [comment]
> There looks to be a couple more occurrences of report in the examples that might need looking at

I left these in on purpose – we still "report errors" (the spec even has a section called "reporting errors") – it's just the construct in code which has changed.

I'll address all your other comments though, they're useful 👍

File: packages/plugin-react-native-global-error-handler/package.json (hunk 1 above)
Comment (bengourley) [reply packages/plugin-react-native-global-error-handler/package.json]: Yeah the explanation is in this commit: https://github.com/bugsnag/bugsnag-js/pull/646/commits/c342609cc603123ad636d4fa200929e93e6e81d5

It was needed to ensure the tests pass.

File: packages/plugin-node-surrounding-code/test/surrounding-code.test.js [hunk 5]
```diff
@@ -66,8 +66,8 @@ describe('plugin: node surrounding code', () => {
     const client = new Client(VALID_NOTIFIER)
 
     client.delivery(client => ({
-      sendReport: (report) => {
-        const evt = report.events[0]
+      sendEvent: (event) => {
+        const evt = event.events[0]
```
[the remark this answers is not in the document]
Comment (bengourley) [reply packages/plugin-node-surrounding-code/test/surrounding-code.test.js]: it's a poor choice of name – the argument should be called `payload`

**Pull request merged**